sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuGetJsonObject.scala

```diff
@@ -136,28 +109,29 @@ case class GpuGetJsonObject(json: Expression, path: Expression)
   override def nullable: Boolean = true
   override def prettyName: String = "get_json_object"
 
-  private var cachedNormalizedPath: Option[Option[String]] = None
+  private var cachedInstructions: 
+      Option[Option[(List[Int], List[String], List[Long])]] = None
 
-  def normalizeJsonPath(path: GpuScalar): Option[String] = {
+  def normalizeJsonPath(path: GpuScalar): Option[(List[Int], List[String], List[Long])] = {
     if (path.isValid) {
       val pathStr = path.getValue.toString()
-      JsonPathParser.parse(pathStr).map(JsonPathParser.normalize)
+      JsonPathParser.parse(pathStr).map(JsonPathParser.splitInstructions)
     } else {
       None
     }
   }
 
   override def doColumnar(lhs: GpuColumnVector, rhs: GpuScalar): ColumnVector = {
-    cachedNormalizedPath.getOrElse {
-      val normalizedPath: Option[String] = normalizeJsonPath(rhs)
-      cachedNormalizedPath = Some(normalizedPath)
-      normalizedPath
+    cachedInstructions.getOrElse {
+      val pathInstructions = normalizeJsonPath(rhs)
+      cachedInstructions = Some(pathInstructions)
+      pathInstructions
     } match {
-      case Some(normalizedStr) => 
-        withResource(Scalar.fromString(normalizedStr)) { scalar =>
-          lhs.getBase().getJSONObject(scalar, 
-              GetJsonObjectOptions.builder().allowSingleQuotes(true).build())
+      case Some(instructions) => instructions match {
+        case (a: List[Int], b: List[String], c: List[Long]) => {
+          JSONUtils.getJsonObject(lhs.getBase, a.toArray, b.toArray, c.toArray)
```

Comment:
Rename a b c to have more meaningful name.
Why not use GPU column view here for a b c as lhs did?
The GPU column view life will remain until `getJsonObject` is done.
In the JNI repo, we can safely refer to the string::view in them.

Reply:
done.